torch/_dynamo/variables/builder.py

```diff
@@ -1433,15 +1433,12 @@ def wrap_tensor(self, value: torch.Tensor):
         ):
             unimplemented("torch.compile does not support strided NestedTensor")
 
-        # Reject sparse, but not coo.
-        # TODO: remove this altogether when non-coo sparsity propagation is ready
-        if is_sparse_any(value) and not value.is_sparse:
-            unimplemented(
-                f"torch.compile does not support sparse Tensor with {value.layout} layout"
-            )
-
         # TODO(pearu,sparse-team) - Add the corresponding SPARSE_TENSOR_MATCH guards
-        if is_sparse_any(value) and value.is_sparse and not self.tx.export:
+        if (
+            isinstance(value, torch.Tensor)
+            and is_sparse_any(value)
+            and not self.tx.export
```

Comment:
I'd prefer it if we can gate this functionality by a config flag, so it's easy for people to fall back to graph breaks if they run into bugs (and I see some unresolved bugs in this PR).

Reply:
I agree currently it is risky (well, I **_know_** given how long this PR has been taken ;-), but don't we want to bite the bullet and make sure the sparse path is made free of bugs? I am afraid special casing this will simply sweep bugs under the carpet.

But your call of course, the change will probably be less risky for me too ;-)

Reply:
What I'm trying to avoid is you getting reverted a month later, which is about how long it will take for this to actually hit prod. If you do want to turn it on by default, at least stage it as two PRs so the config one can be sacrificially reverted while keeping your changes here.

Reply:
Sounds good. Thanks for the advise. I will start with the suggested `capture_sparse_compute = False if is_fbcode() else True`, see if that sticks and then move on to pondering on making it True by default in a (much) later PR